worlds/pokemon_emerald/locations.py

```diff
@@ -173,18 +173,22 @@ def set_free_fly(world: "PokemonEmeraldWorld") -> None:
     # If not enabled, set it to Littleroot Town by default
     fly_location_name = "EVENT_VISITED_LITTLEROOT_TOWN"
     if world.options.free_fly_location:
-        fly_location_name = world.random.choice([
-            "EVENT_VISITED_SLATEPORT_CITY",
-            "EVENT_VISITED_MAUVILLE_CITY",
-            "EVENT_VISITED_VERDANTURF_TOWN",
-            "EVENT_VISITED_FALLARBOR_TOWN",
-            "EVENT_VISITED_LAVARIDGE_TOWN",
-            "EVENT_VISITED_FORTREE_CITY",
-            "EVENT_VISITED_LILYCOVE_CITY",
-            "EVENT_VISITED_MOSSDEEP_CITY",
-            "EVENT_VISITED_SOOTOPOLIS_CITY",
-            "EVENT_VISITED_EVER_GRANDE_CITY",
-        ])
+        free_fly_locations = [
+            "EVENT_VISITED_SLATEPORT_CITY" if "Slateport City" not in world.options.free_fly_blacklist else None,
+            "EVENT_VISITED_MAUVILLE_CITY" if "Mauville City" not in world.options.free_fly_blacklist else None,
+            "EVENT_VISITED_VERDANTURF_TOWN" if "Verdanturf Town" not in world.options.free_fly_blacklist else None,
+            "EVENT_VISITED_FALLARBOR_TOWN" if "Fallarbor Town" not in world.options.free_fly_blacklist else None,
+            "EVENT_VISITED_LAVARIDGE_TOWN" if "Lavaridge Town" not in world.options.free_fly_blacklist else None,
+            "EVENT_VISITED_FORTREE_CITY" if "Fortree City" not in world.options.free_fly_blacklist else None,
+            "EVENT_VISITED_LILYCOVE_CITY" if "Lilycove City" not in world.options.free_fly_blacklist else None,
+            "EVENT_VISITED_MOSSDEEP_CITY" if "Mossdeep City" not in world.options.free_fly_blacklist else None,
+            "EVENT_VISITED_SOOTOPOLIS_CITY" if "Sootopolis City" not in world.options.free_fly_blacklist else None,
+            "EVENT_VISITED_EVER_GRANDE_CITY" if "Ever Grande City" not in world.options.free_fly_blacklist else None,
```

Comment:
I think it would be better to build a dict from the option keys to event names and then turn the option value into a list. Simplifies a little and looks nicer.
```py
my_dict = {
    "Slateport City": "EVENT_VISITED_SLATEPORT_CITY",
    # etc.
}

...

free_fly_locations = sorted(set(my_dict.values) - set(my_dict[city] for city in world.options.free_fly_blacklist.value))
fly_location_name = world.random.choice(free_fly_locations )
```

Something to that effect, I'm writing this entirely in the GH text editor, so I'm not going to guarantee correctness. Make sure to keep the `sorted` for determinism. Otherwise, iterating over a set isn't deterministic.

It'll need a _little_ more code to account for `world.options.free_fly_blacklist.value` containing every possible city, otherwise `random.choice` will complain at you for giving it an empty list.